### ISSUE TITLE
flatpak and snap packages

### ISSUE THREAD

**Issue by nuttyartist** [nuttyartist/notes#501]
We should already have some recipes ready for snap packages. I would like to update the package we currently serve on snapcraft (it looks like version 0.9.0 is still there https://snapcraft.io/notes from 2017). So it would be good to have the automatic process also create artifacts for snaps.

Flathub, with its Flatpak, is growing in popularity and I think we should also upload our app there. 

We can disable our auto-update feature using https://github.com/nuttyartist/notes/pull/474 for both packages intended to be uploaded to the store.

**Comment by guihkx** [comment]
I can handle the Flatpak part.

I already have a Flatpak package of Notes pretty much ready [on a separate repo](https://github.com/guihkx/flatpak-app-notes), so now it's just a matter of:

1. Updating the build steps (from `qmake` to `cmake`).
2. You release v2.1.0.
3. I [submit the app for review](https://github.com/flathub/flathub/wiki/App-Submission) on Flathub.

But I'd like to add the [`.metainfo.xml`](https://github.com/guihkx/flatpak-app-notes/blob/master/io.github.nuttyartist.notes.metainfo.xml) to *this* repository before you release v2.1.0, so it can be shared among other packaging formats (it's pretty much [a Linux standard](https://freedesktop.org/software/appstream/docs/chap-Metadata.html)).

Additionally, if you don't mind, I'd like to first tackle a couple of issues throughout the weekend before v2.1.0 is released:

- [ ] https://github.com/nuttyartist/notes/issues/466
- [ ] https://github.com/nuttyartist/notes/issues/494
- [ ] https://github.com/nuttyartist/notes/discussions/477

**Comment by nuttyartist** [comment]
Perfect!

> Additionally, if you don't mind, I'd like to first tackle a couple of issues, hopefully throughout the weekend, before v2.1.0 is finally released:

Sure. Let's do it.

I skimmed through the files on the separate repo. Looks very good. I'll suggest adding two more screenshots tho.

BTW, the app icons for apps on Flathub look so good. We'll need to up ourselves a notch there. If any of you know a good designer, I will happily pay for a professional-looking icon.

**Comment by guihkx** [comment]
>If any of you know a good designer, I will happily pay for a professional-looking icon.

I don't of anyone, unfortunately. :/

>I'll suggest adding two more screenshots tho.

Agreed. I'm not too great at it, though. I like the second GIF you added to the README, and I think it could yield some great screenshots.

If you still have that notes database (and assuming it's not private), perhaps you could share it here so I can take some new screenshots on Linux?

**Comment by nuttyartist** [comment]
> If you still have that notes database (and assuming it's not private), perhaps you could share it here so I can take some new screenshots on Linux?

Sure!

[notes.db.zip](https://github.com/nuttyartist/notes/files/10889196/notes.db.zip)

**Comment by guihkx** [comment]
That's perfect, thanks! Here some simple screenshots I took on KDE Plasma 5.27 using the Qt 6 AppImage:

![main-window-light-theme](https://user-images.githubusercontent.com/626206/222927528-64941f78-4093-4d3d-8000-fd0dca61980a.png)

![editor-settings](https://user-images.githubusercontent.com/626206/222927531-a52cd062-acd1-49b2-9b58-543ff5772c6e.png)

![main-window-dark-theme](https://user-images.githubusercontent.com/626206/222927543-6c91792f-3fa3-430b-94b7-c2ff92f6ed4d.png)

![mainwindow-sepia-theme](https://user-images.githubusercontent.com/626206/222927546-1d333c84-2d82-4e2d-8c96-a15f42d56af2.png)

Let me know if they look good enough!

**Comment by zjeffer** [comment]
Looks great! 

Maybe we should also add a screenshot of the app with the notes list and/or the folder tree closed. Or maybe we need to create a new GIF that showcases this? The current one in the README uses the older header colors which doesn't look very good.

**Comment by guihkx** [comment]
>Maybe we should also add a screenshot of the app with the notes list and/or the folder tree closed

Good idea. I took a couple of extras screenshots with just the text editor widget visible. I'll add them here:

![editor-only-white](https://user-images.githubusercontent.com/626206/222930768-87a1f5ef-1707-45c3-9322-fb0e64869cc2.png)

![editor-only-dark](https://user-images.githubusercontent.com/626206/222930772-c190637c-5dfc-4962-a2ef-b77eb5218742.png)

![editor-only-sepia](https://user-images.githubusercontent.com/626206/222930775-5808ef81-0d3f-470a-9cfb-5d3d61fd24ef.png)

>Or maybe we need to create a new GIF that showcases this?

The ones in the README could be updated as well, sure. But in this case in specific, we're dealing with screenshots that will go into the [AppData](https://freedesktop.org/software/appstream/docs/chap-Quickstart.html#qsr-app-screenshots-info) file, to showcase the app in Linux app stores, like [Flathub](https://beta.flathub.org/) or the [Snap Store](https://snapcraft.io/notes).

In my understanding, animated GIFs would not be allowed there (or they would just be converted to a static image).

**Comment by nuttyartist** [comment]
Well, our app doesn't look very well in those pictures ):

I'll try to take pictures from Elementary OS, I think it should look better there. If I remember correctly frameless flag looks the best on Ubuntu.

Or maybe, I should create screenshots from macOS? That will be ideal.

**Comment by nuttyartist** [comment]
Let's use these:

![Screen Shot 2023-03-05 at 5 08 29 PM](https://user-images.githubusercontent.com/16375940/222969051-967ac25d-25bd-44ef-8d4d-b128b73245ae.png)
![Screen Shot 2023-03-05 at 5 09 32 PM](https://user-images.githubusercontent.com/16375940/222969053-d240681f-7e8b-4ab6-be34-06df69b61f6c.png)
![Screen Shot 2023-03-05 at 5 09 52 PM](https://user-images.githubusercontent.com/16375940/222969055-c1ff93c8-9c31-427e-ac04-6aafc1e10427.png)
![Screen Shot 2023-03-05 at 5 10 07 PM](https://user-images.githubusercontent.com/16375940/222969056-c364b6a9-39b5-40b8-b0cb-06f022ac1718.png)
![Screen Shot 2023-03-05 at 5 10 45 PM](https://user-images.githubusercontent.com/16375940/222969058-7b41c4ab-2354-4949-a1ce-5ad8751c469b.png)
![Screen Shot 2023-03-05 at 5 11 19 PM](https://user-images.githubusercontent.com/16375940/222969060-a5c5083f-fef8-4e0c-8dcb-9088990b6577.png)
![Screen Shot 2023-03-05 at 5 11 38 PM](https://user-images.githubusercontent.com/16375940/222969061-addf9f1a-2917-4cf1-9afc-20dd4bad2562.png)

**Comment by nuttyartist** [comment]
I suggest this order:
![Screen Shot 2023-03-05 at 5 08 29 PM](https://user-images.githubusercontent.com/16375940/222969217-e68faef4-6ec

![Screen Shot 2023-03-05 at 5 09 32 PM](https://user-images.githubusercontent.com/16375940/222969232-1537ce30-6f81-4f27-a34c-b8479310bd31.png)
9-4a90-9e60-860778039923.png)

![Screen Shot 2023-03-05 at 5 11 38 PM](https://user-images.githubusercontent.com/16375940/222969222-8f30df28-500e-4589-b87c-6e0563ff4ff1.png)

... other images you want to add...

**Comment by guihkx** [comment]
>Well, our app doesn't look very well in those pictures ):

You think so? :(

I think it looks nice... :P

>Or maybe, I should create screenshots from macOS? That will be ideal.

Well, in my opinion, that would be misleading to Linux users, because that's not *really* how the app will look like...

Additionally, as [recommended by the AppStream specification](https://freedesktop.org/software/appstream/docs/chap-Quickstart.html#:~:text=Some%20examples%20of%20good%20and%20bad%20screenshot%20choices%3A), using screenshots that were not taken on Linux is considered "bad":

![image](https://user-images.githubusercontent.com/626206/222969673-6d771c45-13db-465b-847f-822c8aed752a.png)

However, using screenshots taken on elementaryOS would be fine. :)

**Comment by nuttyartist** [comment]
> Additionally, as [the AppStream specification points out](https://freedesktop.org/software/appstream/docs/chap-Quickstart.html#qsr-app-screenshots-info:~:text=Some%20examples%20of%20good%20and%20bad%20screenshot%20choices%3A), using screenshots that were not taken on Linux is considered "bad":

Fair. I will see what I can do.

**Comment by nuttyartist** [comment]
I've tried to run Ubuntu from the live image for the screenshots. But connecting an external USB disk to get the AppImage and database is impossible, and networking drivers are not on by default. Does someone know a workaround?

**Comment by guihkx** [comment]
> But connecting an external USB disk to get the AppImage and database is impossible

Weird, why wouldn't that be possible?

>networking drivers are not on by default

I'm not sure if you're using a laptop and trying to use the internet via Wi-Fi, but I've heard they're usually a mess on Linux... :/

Maybe try via cable if you have one?

**Comment by nuttyartist** [comment]
I think I used both a corrupted one another with a file system compatible only with macOS.

I will try again tonight with a different one.

**Comment by nuttyartist** [comment]
So, no lack on Ubuntu, but your pictures are either as good or better than the ones I took on ElementaryOS. So we can go with this for now.

**Comment by guihkx** [comment]
Alright. :)

Do you still plan to release 2.1.0 today?

**Comment by nuttyartist** [comment]
The only thing left is signing and notarizing the macOS binary. It's a holiday here, so I'll look into that tomorrow.
